woocommerce/wc-admin
# Date Picker: Layout component, presets, and data flow

Depends on https://github.com/woocommerce/woo-dash/pull/63

This code lays out the date picker components, styles and data flow. 

<img width="570" alt="screen shot 2018-05-21 at 8 54 59 pm" src="https://user-images.githubusercontent.com/1922453/40299264-d416f7ba-5d39-11e8-87db-2d98acd28141.png">

<img width="537" alt="screen shot 2018-05-21 at 8 58 48 pm" src="https://user-images.githubusercontent.com/1922453/40299271-da190680-5d39-11e8-9888-4f0ca41ebd76.png">

### Notes
* Initial state (selected tab, preset, compare period, or custom dates) will be gathered from the url so values pre-populated upon opening the datepicker reflect what the user is seeing
* Gutenberg's `<TabPanel />` does not allow control over which tab is initially selected, so I'll make a PR there
* Clicking "Update" simply logs out the internal state, which can be processed into the final output

### Follow up tasks
* Gather params from the url with default values
* Pull request Gutenberg's `<TabPanel />` to allow control over initially selected tab. PR here https://github.com/WordPress/gutenberg/pull/6885
* Create the launcher https://github.com/woocommerce/woo-dash/issues/62
* Create hover states for options and buttons
* Process internal state into an output that an api can consume, ie:
```js
[ 
    { start: '2018-4-15', end: '2018-4-21' },
    { start: '2017-4-15', end: '2017-4-21' }
]
```
### Test 
1. `/wp-admin/admin.php?page=woodash#/analytics`
2. Make sure styles, buttons work
3. Click "Update" and make sure internal state (logged to the console) reflects selection and enough information is present to calculate correct date ranges. "Custom" is not implemented, so that will always be `null`

Fixes https://github.com/woocommerce/woo-dash/issues/58
